apps/pkey.c

```diff
@@ -154,6 +203,20 @@ int pkey_main(int argc, char **argv)
     if (pkey == NULL)
         goto end;
 
+#ifndef OPENSSL_NO_EC
+    if (new_ec_form || new_ec_asn1_flag) {
+        if ((eckey = EVP_PKEY_get0_EC_KEY(pkey)) == NULL) {
+            ERR_print_errors(bio_err);
+            goto end;
+        }
+        if (new_ec_form)
+            EC_KEY_set_conv_form(eckey, ec_form);
+
+        if (new_ec_asn1_flag)
+            EC_KEY_set_asn1_flag(eckey, ec_asn1_flag);
+    }
+#endif
```

Comment:
I wholeheartedly dislike this solution...

... but I understand why this happens, we don't have the necessary control string functions to allow algorithm specific opts like we do in `genpkey` and `pkeyutil`.  Yet another case of incomplete APIs :-/

Reply:
So I think that this will have to be refactored at some point so we don't get this sort of ugly hackery in a program that ideally shouldn't need it.

I would like to see a TODO comment added to this effect

Reply:
I'm not super keen on this either.  I don't like the idea of introducing new command line options that then get long term support when we're intending to remove them soon after.

Adding a `pkeyopt` option to this command is a possibility, although neither of these things appear to be plumbed through to providers currently (i.e. it's more work than we thought).

The third possibility would be to not commit this and rely on the _(deprecated)_ `ec` command for these two until we can return to this.

This seems like the best of the three possibilities but none are great.

Reply:
TODO comment added.

Reply:
What's even more exasperating is that these are text serializing flags...  they have *nothing* to do in groups (that's where they end up).  In OSSL_SERIALIZER terms, they should be passed using `OSSL_SERIALIZER_CTX_set_params()`.  That requires, of course, that we convert the whole `pkey` command to use OSSL_SERIALIZER directly, which actually wouldn't be so bad.  I think we'd get better consistency that way.